subworkflows/local/utils_iridanextexample2/main.nf

```diff
@@ -71,15 +71,28 @@ workflow PIPELINE_INITIALISATION {
     //
     // Create channel from input file provided through params.input
     //
-
+    // Track processed IDs
+    def processedIDs = [] as Set
     Channel
         .fromList(samplesheetToList(params.input, "${projectDir}/assets/schema_input.json"))
         .map {
             meta, fastq_1, fastq_2 ->
+                if (!meta.id) {
+                meta.id = meta.irida_id
+            } else {
+                // Non-alphanumeric characters (excluding _,-,.) will be replaced with "_"
+                meta.id = meta.id.replaceAll(/[^A-Za-z0-9_.\-]/, '_')
+            }
+            // Ensure ID is unique by appending meta.irida_id if needed
+            while (processedIDs.contains(meta.id)) {
+                meta.id = "${meta.id}_${meta.irida_id}"
+            }
+            // Add the ID to the set of processed IDs
+            processedIDs << meta.id
                 if (!fastq_2) {
-                    return [ meta.id, meta + [ single_end:true ], [ fastq_1 ] ]
+                    return [ meta.irida_id, meta + [ single_end:true ], [ fastq_1 ] ]
```

Comment:
Shouldn't this be `return [ meta + [ single_end: true ], [ fastq_1 ] ]`? Or is this to support some functions that exist as part of the nf-core template?

Reply:
Okay. If it's part of nf-core template, then that makes sense. Thanks.